Makefile

```diff
@@ -1,6 +1,39 @@
+.PHONY: make-artifacts-dir generate validate-sigs test coverage install-metrics-binaries lint
+
+export GOLANGCI_LINT_VERSION := v1.62.2
+ifndef GOPATH
+    GOPATH=$(shell go env GOPATH)
+    export GOPATH
+endif
+ifndef ARTIFACTS
+	ARTIFACTS=/tmp/artifacts
+    export ARTIFACTS
+endif
+ifndef COVERAGE_OUTPUT_PATH
+	COVERAGE_OUTPUT_PATH=${ARTIFACTS}/coverage.html
+    export COVERAGE_OUTPUT_PATH
+endif
+
+make-artifacts-dir:
+	mkdir -p ${ARTIFACTS}
+
 generate:
 	go run ./validators/cmd/sigs --dry-run=false
 	go run ./generators/cmd/sigs
 
 validate-sigs:
 	go run ./validators/cmd/sigs
+
+test:
+	go test ./...
+
+coverage: make-artifacts-dir
+	if ! command -V covreport; then go install github.com/cancue/covreport@latest; fi
+	go test ./... -coverprofile=/tmp/coverage.out
+	covreport  -i /tmp/coverage.out -o ${COVERAGE_OUTPUT_PATH}
+
+install-metrics-binaries:
+	if ! command -V golangci-lint; then curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b ${GOPATH}/bin ${GOLANGCI_LINT_VERSION} ; fi
```

Comment:
nit - It would be nicer to install this into a local bin to avoid changing the users version

https://github.com/kubevirt/common-instancetypes/blob/cef52ded0abde7b37564faa619aa745d10b5e335/Makefile#L151-L154